drgn/helpers/linux/idr.py

```diff
@@ -30,9 +98,26 @@ def idr_for_each(idr: Object) -> Iterator[Tuple[int, Object]]:
     :param idr: ``struct idr *``
     :return: Iterator of (index, ``void *``) tuples.
     """
-    try:
-        base = idr.idr_base.value_()
-    except AttributeError:
-        base = 0
-    for index, entry in radix_tree_for_each(idr.idr_rt.address_of_()):
-        yield index + base, entry
+    if idr.prog_.type("struct idr").has_member("idr_rt"):
```

Comment:
In this case, we can follow ["easier to ask for forgiveness than permission"](https://docs.python.org/3/glossary.html#term-EAFP) and assume that `idr_rt` exists, then handle if it doesn't. Something like:

```python3
    # Since Linux kernel commit 0a835c4f090a ("Reimplement IDR and IDA using
    # the radix tree") (in v4.11), IDRs are backed by radix trees.
    try:
        idr_rt = idr.idr_rt
    except AttributeError:
        ... Pre-4.11 version here ...
    else:
        try:
            base = idr.idr_base.value_()
        except AttributeError:
            base = 0
        for index, entry in radix_tree_for_each(idr_rt.address_of_()):
            yield index + base, entry
```